mongodb_consistent_backup/Oplog/Tailer/Tailer.py

```diff
@@ -102,8 +102,8 @@ def stop(self, kill=False, sleep_secs=3):
                     logging.warning("Could not get current optime from PRIMARY! Using now as a stop time")
                     timestamp = Timestamp(int(time()), 0)
 
-                # wait for replication to get in sync
-                while state.get('last_ts') and state.get('last_ts') < timestamp:
+                # wait for replication to get in sync making sure cursor has not been stopped in a race condition
+                while state.get('last_ts') and state.get('last_ts') < timestamp and not self.shards[shard]['thread'].stopped:
```

Comment:
Seems like the first condition is unnecessary if the second depends on it anyway.